src/CoreWCF.UnixDomainSocket/src/CoreWCF/Security/UnixDomainSocketInterop.cs

```diff
@@ -0,0 +1,144 @@
+{
+    internal static class UnixDomainSocketInterop
+    {
+       
+        internal static bool TryGetCredentials(this Socket socket, out uint processId, out uint userId, out uint groupId)
+        {
+            try
+            {
+                if (OperatingSystem.IsWindows())
+                    throw new PlatformNotSupportedException();
+
+                Span<uint> ucred = stackalloc uint[3];
+                int bytesWritten;
+                if (OperatingSystem.IsMacOS())
+                {
+                    bytesWritten = socket.GetRawSocketOption(0, 2, MemoryMarshal.AsBytes(ucred)); // get processid
+                    processId = ucred[0];
+                    userId = 0;
+                    groupId = 0;
+                    return true;
+                }
+                else
+                {
+                    const int SOL_SOCKET = 1;
+                    const int SO_PEERCRED = 17;
+                    bytesWritten = socket.GetRawSocketOption(SOL_SOCKET, SO_PEERCRED, MemoryMarshal.AsBytes(ucred));
+                }
+                processId = ucred[0];
+                userId = ucred[1];
+                groupId = ucred[2];
+                return bytesWritten == (ucred.Length * sizeof(uint));
+            }
+            catch(Exception ex)
+            {
+                throw;
+            }
+        }
+
+    }
+
+    internal class NativeSysCall
+    {
+        const string LIBC = "libc";
+
+        [DllImport(LIBC, SetLastError = true)]
+        private static extern IntPtr getgrgid(uint name);
+
+        [DllImport(LIBC, SetLastError = true)]
+        private static extern IntPtr getpwuid(uint uid);
+
+    //    [DllImport(LIBC, SetLastError = true)]
+  //      private static extern IntPtr getsockopt(int socket, int level, int option_name,
+//void* restrict option_value, socklen_t* restrict option_len);
+
+
+        internal static UserInfo? GetUserInfo(uint uid)
+        {
+            var result = NativeSysCall.getpwuid(uid);
+            if (result != IntPtr.Zero) return new UserInfo(Marshal.PtrToStructure<NativeSysCall.Passwd>(result));
+            return null;
+        }
+
+        internal static GroupInfo? GetGroupInfo(uint grpId)
+        {
+            var result = NativeSysCall.getgrgid(grpId);
+            if (result != IntPtr.Zero) return new GroupInfo(Marshal.PtrToStructure<NativeSysCall.Group>(result));
+            return null;
+        }
+
+        internal struct Group
+        {
+            public string Name;
+            public string Password;
+            public uint Gid;
+            public IntPtr Members;
+
+        }
+
+        internal struct Passwd
+        {
+            public string Name;
+            public string Password;
+            public uint Uid;
+            public uint Gid;
+            public string GECOS;
+            public string Directory;
+            public string Shell;
+
+        }
+
+    }
+    internal sealed class GroupInfo
+    {
```

Comment:
Needs blank line between types. Also there's an extra blank line before the properties. It's also normal to place properties after the constructor